src/vstest.console/CommandLine/TestCaseFilter.cs

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.CommandLine
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Microsoft.VisualStudio.TestPlatform.Common.Filtering;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+    
+    /// <summary>
+    /// Unit test filtering
+    /// </summary>
+    public class TestCaseFilter : ITestCaseFilter
```

Comment:
Suggest to make this a private class within `ListFullyQualifiedTestsArgumentExecutor`. Even static functions are good too.

Reply:
Done